plugins/csp-demo-node-editor/src/nodes/NumberNode.cpp

```diff
@@ -0,0 +1,71 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// SPDX-FileCopyrightText: German Aerospace Center (DLR) <cosmoscout@dlr.de>
+// SPDX-License-Identifier: MIT
+
+#include "NumberNode.hpp"
+
+#include "../../../../src/cs-utils/filesystem.hpp"
+#include "../../../../src/cs-utils/utils.hpp"
+
+namespace csp::demonodeeditor {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const std::string NumberNode::sName = "Number";
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+std::string NumberNode::sSource() {
+  return cs::utils::filesystem::loadToString(
+      "../share/resources/nodes/csp-demo-node-editor/NumberNode.js");
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+std::unique_ptr<NumberNode> NumberNode::sCreate() {
+  return std::make_unique<NumberNode>();
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+std::string const& NumberNode::getName() const {
+  return sName;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void NumberNode::process() {
+  writeOutput("value", mValue);
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void NumberNode::onMessageFromJS(nlohmann::json const& message) {
+
+  // The CosmoScout.sendMessagetoCPP() method sends the current value.
+  mValue = message;
+
+  // Whenever the value, we write it to the output socket by calling the process() method.
```

Comment:
I think a word is missing in this sentence.

Reply:
Could be :sweat_smile: